io/net/virt-net/veth_dlpar.py

```diff
@@ -0,0 +1,168 @@
+        output = self.run_command(cmd)
+        for line in output.splitlines():
+            if "SEA" in line:
+                self.sea = line.split()[-1]
+        self.log.info(self.sea)
+        local = LocalHost()
+        self.networkinterface = NetworkInterface(self.interface, local)
+        if self.networkinterface.ping_check(self.peer_ip, count=5) is not None:
+            self.fail("peer connection is failed")
+
+    def login(self, ipaddr, username, password):
+        '''
+        SSH Login method for remote server
+        '''
+        pxh = pxssh.pxssh(encoding='utf-8')
+        # Work-around for old pxssh not having options= parameter
+        pxh.SSH_OPTS = "%s  -o 'StrictHostKeyChecking=no'" % pxh.SSH_OPTS
+        pxh.SSH_OPTS = "%s  -o 'UserKnownHostsFile /dev/null' " % pxh.SSH_OPTS
+        pxh.force_password = True
+
+        pxh.login(ipaddr, username, password)
+        pxh.sendline()
+        pxh.prompt(timeout=60)
+        pxh.sendline('exec bash --norc --noprofile')
+        pxh.prompt(timeout=60)
+        # Ubuntu likes to be "helpful" and alias grep to
+        # include color, which isn't helpful at all. So let's
+        # go back to absolutely no messing around with the shell
+        pxh.set_unique_prompt()
+        pxh.prompt(timeout=60)
+        self.pxssh = pxh
+
+    def run_command(self, command, timeout=300):
+        '''
+        SSH Run command method for running commands on remote server
+        '''
+        self.log.info("Running the command on peer lpar: %s", command)
+        if not hasattr(self, 'pxssh'):
+            self.fail("SSH Console setup is not yet done")
+        con = self.pxssh
+        con.sendline(command)
+        con.expect("\n")  # from us
+        con.expect(con.PROMPT, timeout=timeout)
+        output = "".join(con.before)
+        con.sendline("echo $?")
+        con.prompt(timeout)
+        exitcode = int(''.join(con.before.splitlines()[1:]))
+        if exitcode != 0:
+            raise CommandFailed(command, output, exitcode)
+        return output
+
+    def veth_dlpar_remove(self):
+        '''
+        veth dlpar remove operation
+        '''
+        cmd = "rmdev -l %s" % self.sea
+        cmd_l = "echo \"%s\" | oem_setup_env" % cmd
+        try:
+            output = self.run_command(cmd_l)
+            self.log.info(output)
+        except CommandFailed as cf:
+            self.fail("failed dlpar remove operation as %s" % cf)
+
+    def veth_dlpar_add(self):
+        '''
+        veth dlpar add operation
+        '''
+        cmd = "mkdev -l %s" % self.sea
+        cmd_l = "echo \"%s\" | oem_setup_env" % cmd
+        try:
+            output = self.run_command(cmd_l)
+            self.log.info(output)
+        except CommandFailed as cf:
+            self.fail("Failed dlpar add operation as %s" % cf)
+        if self.networkinterface.ping_check(self.peer_ip, count=5) is not None:
+            self.fail("ping failed after add operation")
```

Comment:
This should be done in the `test_dlpar()` or as a separate function, not inside `veth_dlpar_add()`

Reply:
ok